AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Sync/SubscriptionSync/IncomingAsyncSubscriptionEventPublisher.swift

```diff
@@ -81,56 +81,47 @@ final class IncomingAsyncSubscriptionEventPublisher: Cancellable {
 
     func onCreateListenerHandler(event: Event) {
         log.verbose("onCreateListener: \(event)")
-        if case .inProcess(.connection) = event {
-            let onCreateConnectionOp = CancelAwareBlockOperation {
-                if case .inProcess(.connection(.connected)) = event {
-                    self.onCreateConnected = true
-                } else if case .inProcess(.connection(.disconnected)) = event {
-                    self.onCreateConnected = false
-                }
-                if self.combinedConnectionStatus {
-                    self.incomingSubscriptionEvents.send(event)
-                }
-            }
-            self.connectionStatusQueue.addOperation(onCreateConnectionOp)
-        } else {
-            incomingSubscriptionEvents.send(event)
+        let onCreateConnectionOp = CancelAwareBlockOperation {
+            self.onCreateConnected = self.determineConnectionStatusOn(event: event)
+            self.sendConnectionEventIfConnected(event: event)
         }
+        genericListenerHandler(event: event, cancelAwareBlock: onCreateConnectionOp)
     }
-    
+
     func onUpdateListenerHandler(event: Event) {
         log.verbose("onUpdateListener: \(event)")
-        if case .inProcess(.connection) = event {
-            let onUpdateConnectionOp = CancelAwareBlockOperation {
-                if case .inProcess(.connection(.connected)) = event {
-                    self.onUpdateConnected = true
-                } else if case .inProcess(.connection(.disconnected)) = event {
-                    self.onUpdateConnected = false
-                }
-                if self.combinedConnectionStatus {
-                    self.incomingSubscriptionEvents.send(event)
-                }
-            }
-            self.connectionStatusQueue.addOperation(onUpdateConnectionOp)
-        } else {
-            incomingSubscriptionEvents.send(event)
+        let onUpdateConnectionOp = CancelAwareBlockOperation {
+            self.onUpdateConnected = self.determineConnectionStatusOn(event: event)
+            self.sendConnectionEventIfConnected(event: event)
         }
+        genericListenerHandler(event: event, cancelAwareBlock: onUpdateConnectionOp)
     }
 
     func onDeleteListenerHandler(event: Event) {
         log.verbose("onDeleteListener: \(event)")
+        let onDeleteConnectionOp = CancelAwareBlockOperation {
+            self.onDeleteConnected = self.determineConnectionStatusOn(event: event)
+            self.sendConnectionEventIfConnected(event: event)
+        }
+        genericListenerHandler(event: event, cancelAwareBlock: onDeleteConnectionOp)
+    }
+
+    func determineConnectionStatusOn(event: Event) -> Bool {
```

Comment:
For Boolean return types, prefer `isXXX(for event: Event)` or similar naming that [asserts something about the receiver](https://swift.org/documentation/api-design-guidelines/#boolean-assertions) and reads like an English question. For this, maybe something like `isConnected(for event: Event)` or `isConnectionValid(for event: Event)`?

Reply:
thanks for calling this out, changed to:
`isConnectionStatusConnected(for event: Event)`